martypy/ClientMV2.py

```diff
@@ -407,6 +408,99 @@ def is_conn_ready(self) -> bool:
             return False
         return self._initComplete and (self.lastSubscribedMsgTime is not None)
 
+    def _is_valid_disco_addon(self, add_on: str) -> bool:
+        disco_type_codes = {"00000087","00000088","00000089"}
+        for attached_add_on in self.get_add_ons_status().values():
+            if type(attached_add_on) == dict and attached_add_on['name'] == add_on:
+                if attached_add_on['whoAmITypeCode'] in disco_type_codes:
+                    return True
+                else:
+                    raise MartyCommandException(f"The add on name: '{add_on}' is not a valid disco add on. "
+                                                "Please check the add on name in the scratch app -> configure -> add ons")
+        raise MartyCommandException(f"The add on name '{add_on}' is not a valid add on. Please check the add on "
+                                    "name in the scratch app -> configure -> add ons")
+
+    def _valid_hex(self, color_hex: str) -> bool:
+        hex_pattern = re.compile("^([A-Fa-f0-9]{6})$")
+        return bool(hex_pattern.match(color_hex))
```

Comment:
Might be worth moving this just above `_parse_color_hex` since that is the only place where it's used?